### PR TITLE
Avoid switching GC mode in COM's HasValidTarget

### LINKED ISSUE #62063: Interop ComWrappersTests failures
Noticed in pri1 run on helix - https://helix.dot.net/api/2019-06-17/jobs/4389830b-bb7a-4316-8d83-fd59f0b90036/workitems/Interop/console

```
  Discovered:  Interop.COM.XUnitWrapper (found 25 test cases)
  Starting:    Interop.COM.XUnitWrapper (parallel test collections = on, max threads = 8)
    Interop\COM\ComWrappers\API\ComWrappersTests\ComWrappersTests.cmd [FAIL]
      
      Assert failure(PID 15768 [0x00003d98], Thread: 14976 [0x3a80]): pThread
      
      CORECLR! InteropLibImports::HasValidTarget + 0x4C (0x00007ffc`666457f4)
      CORECLR! TrackerTarget_QueryInterface + 0x9C (0x00007ffc`66848a8c)
      MOCKREFERENCETRACKERRUNTIME! UpdateTestObjectAsIDispatch + 0x1F5B8 (0x00007ffc`64d7b1b8)
      MOCKREFERENCETRACKERRUNTIME! UpdateTestObjectAsIDispatch + 0x1EA80 (0x00007ffc`64d7a680)
      CORECLR! TrackerObjectManager::BeginReferenceTracking + 0xEC (0x00007ffc`6684a42c)
      CORECLR! InteropLib::Com::BeginExternalObjectReferenceTracking + 0x14 (0x00007ffc`668481c4)
      CORECLR! ComWrappersNative::OnFullGCStarted + 0xF8 (0x00007ffc`66645c98)
      CORECLR! standalone::GCToEEInterface::GcStartWork + 0x28 (0x00007ffc`66641ed8)
      CLRGC! SVR::gc_heap::garbage_collect + 0x584 (0x00007ffc`6ebb286c)
      CLRGC! SVR::gc_heap::gc_thread_function + 0x2E8 (0x00007ffc`6ebb5220)
          File: D:\a\_work\1\s\src\coreclr\vm\threads.inl Line: 43
          Image: D:\h\w\AF030992\p\corerun.exe
      
      
      Return code:      1
      Raw output file:      D:\h\w\AF030992\w\A107096A\uploads\Reports\Interop.COM\ComWrappers\API\ComWrappersTests\ComWrappersTests.output.txt
      Raw output:
      BEGIN EXECUTION
       "D:\h\w\AF030992\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  ComWrappersTests.dll 
      Running ValidateComInterfaceCreation...
      Running ValidateComInterfaceCreationRoundTrip...
      Running ValidateFallbackQueryInterface...
      Running ValidateCreateObjectCachingScenario...
      Running ValidateWrappersInstanceIsolation...
      Running ValidatePrecreatedExternalWrapper...
      Running ValidateExternalWrapperCacheCleanUp...
      Running ValidateSuppliedInnerNotAggregation...
      Running ValidateIUnknownImpls...
      Running ValidateBadComWrapperImpl...
      Running ValidateRuntimeTrackerScenario...
      Expected: 100
      Actual: -1073740286
      END EXECUTION - FAILED
      FAILED
      Test Harness Exitcode is : 1
      To run the test:
      > set CORE_ROOT=D:\h\w\AF030992\p
      > D:\h\w\AF030992\w\A107096A\e\Interop\COM\ComWrappers\API\ComWrappersTests\ComWrappersTests.cmd
      Expected: True
      Actual:   False
      Stack Trace:
           at Interop_COM._ComWrappers_API_ComWrappersTests_ComWrappersTests_._ComWrappers_API_ComWrappersTests_ComWrappersTests_cmd()
      Output:
        
        Assert failure(PID 15768 [0x00003d98], Thread: 14976 [0x3a80]): pThread
        
        CORECLR! InteropLibImports::HasValidTarget + 0x4C (0x00007ffc`666457f4)
        CORECLR! TrackerTarget_QueryInterface + 0x9C (0x00007ffc`66848a8c)
        MOCKREFERENCETRACKERRUNTIME! UpdateTestObjectAsIDispatch + 0x1F5B8 (0x00007ffc`64d7b1b8)
        MOCKREFERENCETRACKERRUNTIME! UpdateTestObjectAsIDispatch + 0x1EA80 (0x00007ffc`64d7a680)
        CORECLR! TrackerObjectManager::BeginReferenceTracking + 0xEC (0x00007ffc`6684a42c)
        CORECLR! InteropLib::Com::BeginExternalObjectReferenceTracking + 0x14 (0x00007ffc`668481c4)
        CORECLR! ComWrappersNative::OnFullGCStarted + 0xF8 (0x00007ffc`66645c98)
        CORECLR! standalone::GCToEEInterface::GcStartWork + 0x28 (0x00007ffc`66641ed8)
        CLRGC! SVR::gc_heap::garbage_collect + 0x584 (0x00007ffc`6ebb286c)
        CLRGC! SVR::gc_heap::gc_thread_function + 0x2E8 (0x00007ffc`6ebb5220)
            File: D:\a\_work\1\s\src\coreclr\vm\threads.inl Line: 43
            Image: D:\h\w\AF030992\p\corerun.exe
        
        
        Return code:      1
        Raw output file:      D:\h\w\AF030992\w\A107096A\uploads\Reports\Interop.COM\ComWrappers\API\ComWrappersTests\ComWrappersTests.output.txt
        Raw output:
        BEGIN EXECUTION
         "D:\h\w\AF030992\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  ComWrappersTests.dll 
        Running ValidateComInterfaceCreation...
        Running ValidateComInterfaceCreationRoundTrip...
        Running ValidateFallbackQueryInterface...
        Running ValidateCreateObjectCachingScenario...
        Running ValidateWrappersInstanceIsolation...
        Running ValidatePrecreatedExternalWrapper...
        Running ValidateExternalWrapperCacheCleanUp...
        Running ValidateSuppliedInnerNotAggregation...
        Running ValidateIUnknownImpls...
        Running ValidateBadComWrapperImpl...
        Running ValidateRuntimeTrackerScenario...
        Expected: 100
        Actual: -1073740286
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        To run the test:
        > set CORE_ROOT=D:\h\w\AF030992\p
        > D:\h\w\AF030992\w\A107096A\e\Interop\COM\ComWrappers\API\ComWrappersTests\ComWrappersTests.cmd
    Interop\COM\ComWrappers\API\ComWrappersTestsBuiltInComDisabled\ComWrappersTestsBuiltInComDisabled.cmd [FAIL]
      
      Assert failure(PID 11576 [0x00002d38], Thread: 16036 [0x3ea4]): pThread
      
      CORECLR! InteropLibImports::HasValidTarget + 0x4C (0x00007ffc`666457f4)
      CORECLR! TrackerTarget_QueryInterface + 0x9C (0x00007ffc`66848a8c)
      MOCKREFERENCETRACKERRUNTIME! UpdateTestObjectAsIDispatch + 0x1F5B8 (0x00007ffc`6b4bb1b8)
      MOCKREFERENCETRACKERRUNTIME! UpdateTestObjectAsIDispatch + 0x1EA80 (0x00007ffc`6b4ba680)
      CORECLR! TrackerObjectManager::BeginReferenceTracking + 0xEC (0x00007ffc`6684a42c)
      CORECLR! InteropLib::Com::BeginExternalObjectReferenceTracking + 0x14 (0x00007ffc`668481c4)
      CORECLR! ComWrappersNative::OnFullGCStarted + 0xF8 (0x00007ffc`66645c98)
      CORECLR! standalone::GCToEEInterface::GcStartWork + 0x28 (0x00007ffc`66641ed8)
      CLRGC! SVR::gc_heap::garbage_collect + 0x584 (0x00007ffc`6ebb286c)
      CLRGC! SVR::gc_heap::gc_thread_function + 0x2E8 (0x00007ffc`6ebb5220)
          File: D:\a\_work\1\s\src\coreclr\vm\threads.inl Line: 43
          Image: D:\h\w\AF030992\p\corerun.exe
      
      
      Return code:      1
```

### PR DESCRIPTION
**Pull request by am11**
Fixes #62063.

### CODE REVIEW

**Comment by am11** [comment]
cc @jkotas, can we run the [`gc-standalone`](https://dev.azure.com/dnceng/public/_build?definitionId=1017) pipeline for this PR, as it is the one where failing test was discovered? (this pipeline is failing for some time)

File: src/coreclr/vm/interoplibinterface_comwrappers.cpp [hunk 1]
```diff
@@ -1169,11 +1169,7 @@ namespace InteropLibImports
         bool isValid = false;
         ::OBJECTHANDLE objectHandle = static_cast<::OBJECTHANDLE>(handle);
 
-        {
-            // Switch to cooperative mode so the handle can be safely inspected.
-            GCX_COOP_THREAD_EXISTS(GET_THREAD());
-            isValid = ObjectFromHandle(objectHandle) != NULL;
-        }
+        isValid = ObjectHandleIsNull(objectHandle) == TRUE;
```
Comment (jkotas) [inline src/coreclr/vm/interoplibinterface_comwrappers.cpp:1172]:
```suggestion
        isValid = ObjectHandleIsNull(objectHandle) != FALSE;
```
It is bug prone to compare `BOOL` with `TRUE` and some static analysis tools will flag it as a problem.
For example https://pvs-studio.com/en/docs/warnings/v676/

**Comment by jkotas** [comment]
> can we run the gc-standalone pipeline for this PR

Triggered

**Review by jkotas — APPROVED** [review]
Thank you!

**Comment by am11** [comment]
COM wrapper test is not failing anymore: https://dev.azure.com/dnceng/public/_build/results?buildId=1487089&view=ms.vss-test-web.build-test-results-tab (compared with the previous run https://dev.azure.com/dnceng/public/_build/results?buildId=1486145&view=ms.vss-test-web.build-test-results-tab).

For remaining failures, I have opened #62083 and #62084. RuntimeConfiguration test is a new one, so I reported to @AaronRobinsonMSFT directly: https://github.com/dotnet/runtime/pull/61950#discussion_r757621079.

**Pull request merged**